File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:

        dummy = ListNode(-1)
        dummy.next = head

        st = set(nums)

        prv = dummy
        curr = head

        while curr:
            if curr.val in st:
                prv.next = curr.next
            else:
                prv = curr

            curr = curr.next
        
        return dummy.next
        

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) 
    {

        Set<Integer> st = new HashSet<>();
        for(int i : nums)
        {
            st.add(i);
        } 

        ListNode dummy = new ListNode(-1);
        dummy.next = head;

        ListNode prev = dummy;
        ListNode curr = head;

        while ( curr != null)
        {
            if ( st.contains(curr.val))
            {
                prev.next = curr.next;
            }
            else prev = curr;
            curr = curr.next;
        }

        return dummy.next;

        
    }
}